### ISSUE TITLE
Game takes an absurd amount of time to start/ load

### ISSUE THREAD

**Issue by guianselmi** [ValveSoftware/Dota-2#1968]
# Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/guianselmi/5894ccdc116e3116d2c43988a02d2508): 
* Have you checked for system updates?: [Yes]
* Are you using the latest stable video driver available for your system? [Yes]
* Have you verified the game files?: [Yes]

# Please describe your issue in as much detail as possible:
I have been playing Dota on Ubuntu for a long time already, including last week. To help set a timeframe of when this issue started: this past weekend, there was the Nemestice update where the following was implemented:

- Start up banner informing the time left for the Nemestice battle pass
- New mainscreen background image (screenshot not mine)
![bg](https://user-images.githubusercontent.com/4984657/129056710-cf2efcb9-1a44-456d-aad2-d99b9d177c65.jpg)

... besides other many automatic downloads of Vulkan caching, and other Steamworks stuff that auto downloaded when I installed the game Ghostrunner, which was free for a short time.

## Expected result

Starting Dota 2 takes only a few seconds, as it usually was (almost instant).

## Current result

The game now takes at least 17m30s to open, while the Steam interface says it is "Running". I tested it 2 times right after starting Steam, and oddly enough they both took exactly 17m30s for the game to finally open. After it opened, I closed the game and tried to open it again: this time, I gave up after it went past 25 minutes...

Please note: 

- The only "Launch option" I am currently using is `-novid`.
- I uninstalled the game via Steam's UI, and reinstalled it again. Same broken result. I did not remove anything manually though, like the `/.steam/debian-installation/steamapps/shadercache` dir, which was tempting (it currently has 2,4 GB)...
- CS:GO opens **normally**.

# Steps for reproducing this issue:

1. Open Steam
2. Find Dota 2 in your Library
3. Click Play
4. Notice how the game is taking an absurd amount of time to start.

**Comment by guianselmi** [comment]
Relevant: after cleaning up my launch options and keeping only `-gl`, the game opened again normally.

Why is that? I have been using vulkan for a long time, even **without** adding that to the launch options (my launch options have always been only `-novid`). Also, I want to use the "FX super resolution" (I forgot how it's called), which works with Vulkan and not OpenGL... I could not find the Vulkan DLC to download. How can I get it again and start fresh? I want my game to use Vulkan.

If the launch options does not have anything, the game does not open. It is only opening if using `-gl`. I thought vulkan was default...?

**Comment by DGauze** [comment]
Just got this issue this morning, hopped on my pc to play and all I get is a black screen after the Dota logo briefly appears on startup. I did not wait the whole 17 minutes, however. What's extra weird is that when the menu music plays, it isn't even my selected music pack. My previous launch option was `-vulkan`, and it only works properly with the `-gl` option as guianselmi stated. This is with *no* system updates in between launches, I kept my system on overnight for a download and when I got on to play I encountered the issue.

Also reinstalled the game and it didn't fix it, under the assumption that a config file somewhere got borked. I also timeshifted to a previous state of my machine (a few days ago before a system update on Arch) to see if somehow that would help but it didn't.

My gpu is an RX 6900xt, if that's relevant. All my other games work perfectly, except for Hades oddly enough which also an issue initializing the vulkan loader.

**Comment by rpayanm** [comment]
I had the same issue, I had to change from Vulkan to OpenGL, now It is working fine.

**Comment by cxspxr** [comment]
same here, ryzen 2500u

**Comment by jazztickets** [comment]
Mine takes about a minute to load. Once you're in the main menu, the FPS continues to be around 1-2. If you can get into the demo hero mode, everything clears up. Happens on two different amdgpu systems. (5700xt, rx570)

Video:
https://user-images.githubusercontent.com/425930/130304094-d9939df9-92ee-4c62-90b9-2fe0f61b8452.mp4

System:
https://gist.githubusercontent.com/jazztickets/f07f40cf0896a257b911325fc21f5e1a/raw/3a12ade4bb9f236fef6702e6fe01bc732d4266ab/main%2520system%2520info

**Comment by mfbsouza** [comment]
I'm having the same issue as in @jazztickets video when using vulkan.

my system: Arch Linux, linux-zen 5.13.12-zen1-1-zen, RX580 with RADV 21.2.1

I'm wondering if this is a dota bug or this issue is related with the RADV 21.2.1 driver since @jazztickets is also running with it. although i tested other native vulkan games and they are running fine.

**Comment by DGauze** [comment]
Since the recent mini-updates to Dota my game instead of just being stuck at a blank screen, thankfully now crashes so I can see a log of what's going on exactly when it does. Here is the full output of `steam --debug` from launching steam to the actual game attempting to load: https://pastebin.com/NFuVnUAi

Specifically line 531 in the pastebin may be relevant? It reads as:
`/home/dgauze/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh: line 109: 27412 Segmentation fault      (core dumped)`

Confusing cause line 109 simply a `Done` command. Perhaps someone who can parse logs better than I can chime in. 

Edit: Seems my issue wasn't related, yet had similar symptoms.

**Comment by Quickstick4** [comment]
I'm also having the same issue using Vulkan. Once the game finally gets past being at 1-2 FPS in the main menu, it returns to 240 FPS. 
Getting into a normal games always times out. When spectating, once IN the game it performs absolutely fine. Definitely appears to be something to do with loading.  

OS: Ubuntu 21.04
KERNEL: 5.11.0-31-generic
CPU: AMD Ryzen 5 5600X 6-Core
GPU: AMD AMD Radeon RX 6900 XT 
(SIENNA_CICHLID, DRM 3.40.0, 5.11.0-31-generic, LLVM 12.0.1)
GPU DRIVER: 4.6 Mesa 21.2.1 - kisak-mesa PPARAM: 32 GB

**Comment by jazztickets** [comment]
Definitely a problem with amdgpu and mesa 21.2.1. Downgrading {lib32-mesa,lib32-vulkan-radeon,libva-mesa-driver,mesa,vulkan-mesa-layers,vulkan-radeon} to 21.1.6 helps, but the game still stutters a little bit when loading.

I tried it on my nvidia gtx 970 system and it works as intended (even better than amdgpu + 21.1.6).

Here's a video of mesa 21.1.6 on my 5700xt system:
https://user-images.githubusercontent.com/425930/130325908-4b919e65-5f32-4416-9006-d46da13b93e0.mp4

**Comment by skafdan** [comment]
GPU information 
```
03:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT] (rev c1)
	Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] Device 0b36
	Kernel driver in use: amdgpu
	Kernel modules: amdgpu
```
amdgpu version : `21.0.0`

I have also been experiencing the very long load time. 

If you launch steam from the terminal you can notice that during the "hang"/loading it says 
```
failed to dlopen "libtensorflow.so" error=libtensorflow.so: cannot open shared object file: No such file or directory
```

Also during the pick phase their are moments of extreme fps drop which the output is 
```
ERROR: ld.so: object '/home/dan/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
```
Kernel is `5.10.60-1 ` manjaro

**Comment by regulator-g** [comment]
> Definitely a problem with amdgpu and mesa 21.2.1


In another thread an nVidia user has the same issue

seems to only happen with vulkan, openGL is ok

**Comment by lennie420** [comment]
This seems to be resolved for me? Not sure if there was a update to game? I am currently running mesa-git built a few days and hadn't played Dota in about 6 days. Today I played and all seems to be working alright.

edit: nevermind, I downgraded to stable version on Arch and got the lag. Reinstalled mesa-git (rebuilt) and got the lag again... Weird.

edit 2: Okay after restarting Dota now no lag when starting up. Hmm...

**Comment by guianselmi** [comment]
>This seems to be resolved

Not fixed yet. I just tried using `-vulkan`, and the problem returned. So far, it only works with `-gl`. I miss being able to use vulkan and FX super resolution...

**Comment by mfbsouza** [comment]
Since i got this problem i downgraded Mesa and RADV to 21.1.6 so that i can play using vulkan API. It's not completely fine cause i still got some minor loading stutters, a ton of gpu usage in menu and visual bugs in the hero level. But at least it's playable. Here is a screenshot i took of the visual bug:
https://user-images.githubusercontent.com/31019638/132548506-2672cf77-8fca-4880-895b-7770c882d142.png

EDIT: So my dota just had an update and fixed the visual issue that i posted above but the minor stutters and high gpu usage in menu is still present

**Comment by mfbsouza** [comment]
Yesterday Mesa posted a new release (21.2.2) with a lot of fixes. I just updated it in my system (Arch Linux) and got rid with most of the issues that i was having in Dota, including the huge amount of time to load assets. So this issue is now resolved to me.

**Comment by Quickstick4** [comment]
New Mesa drivers fixed the issue for me as well.

**Comment by kisak-valve** [comment]
Blind guess that this issue was related to https://cgit.freedesktop.org/mesa/mesa/commit/?id=d2d642cc014, which is fixed in mesa 21.2.2 by https://cgit.freedesktop.org/mesa/mesa/commit/?id=8de60a1654a875d0b55512a1508f541f706f9b97. If it is, then this issue would be limited to mesa 21.2.1, 21.1.8, and 21.1.7.

**Comment by Saladin1812** [comment]
Mesa 21.2.2 update fixed the following issues #1968 #1970 #1971 for me

**Comment by regulator-g** [comment]
Edit: all good after Mesa 21.2.2 update

**Comment by guianselmi** [comment]
After the comments above, I decided to retry launching the game **without** forcing it to use `-gl`. Somehow, it opened normally. 
Note: it seems to be something fixed on Dota's side, instead of Mesa's. I keep my Ubuntu up to date, but Mesa is still `Mesa 21.0.3`.

Current settings:
- "Steam Settings > Shader Pre-Caching > Enable Shader Pre-Caching" **set**. (currently 831 MB pre-cached)
- "Steam Settings > Shader Pre-Caching > Allow background processing of Vulkan shaders" **set**.
- "Steam Settings > Steam Play > Enable Steam Play for supported titles" **set**
- "Steam Settings > Steam Play > Enable Steam Play for all other titles" **set**
- "Steam Settings > Steam Play > Run other titles with" **Proton 6.3-6**
- Game's launch options: only `-novid`
- Game's compatibility: "Force the use of a specific Steam Play compatibility tool" **unset**.

I am running `Ubuntu 20.04.3 LTS (64 bit)`, still with `Mesa 21.0.3` (according to "Steam Help > System Information" and `glxinfo| grep Mesa`). I don't even know how one would update Mesa in an official way and still use Ubuntu LTS, but now the game at least opens instantly and let's me use vulkan, which by the way was automatically set when it opened. Now, I can decrease the "Game Screen Render Quality" to 99% and activate "FidelityFX Super Resolution".

Should it have been fixed even without the new Mesa? Or, is it something else?

tl;dr: Mesa not updated, but game now runs with vulkan. Playing and spectating works fine.

**Comment by guianselmi** [comment]
I am closing this issue. Please feel free to reopen it in case it is still relevant, @kisak-valve .
Thank you very much
